### PR TITLE
Improve user delete and handle errors

### PR DESCRIPTION
**Pull request by antima-gupta**
### Description

- Added specs for user delete fixes.

### Issues Resolved

#36 

### Check List

- [ ] New functionality includes tests
- [ ] All tests pass
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [ ] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

**Comment by antima-gupta** [comment]
@tylercloke I tried to rebase my branch with the master but it seems `lib/chef/knife/opc_user_delete.rb` having merge conflicts. Attaching conflict file [opc_user_delete_conflict.txt](https://github.com/chef/knife-opc/files/5033421/opc_user_delete_conflict.txt). Could you please rebase [PR](https://github.com/chef/knife-opc/pull/36), so that I can rebase this?

**Comment by tas50** [comment]
@antima-gupta This needs to get rebased off master. I think that may fix the specs not actually running in Buildkite

**Comment by antima-gupta** [comment]
@tas50 I have rebase the PR. Please have a look.

**Comment by antima-gupta** [comment]
@tas50 I have rebase this, Could you please have a look and let me know if any changes required?

File: lib/chef/knife/opc_user_delete.rb [hunk 1]
```diff
@@ -19,126 +19,38 @@
...
-
-    option :no_disassociate_user,
-      long: "--no-disassociate-user",
-      short: "-d",
-      description: "Don't disassociate the user first"
-
-    option :remove_from_admin_groups,
-      long:  "--remove-from-admin-groups",
-      short:  "-R",
-      description: "If the user is a member of any org admin groups, attempt to remove from those groups. Ignored if --no-disassociate-user is set."
```
Comment (lamont-granquist) [inline lib/chef/knife/opc_user_delete.rb:33]: this should not be getting removed, and a bunch of other stuff in here is not part of the other PR which is being rebased.

**Comment by tas50** [comment]
@antima-gupta The @tylercloke's old PR here came before @marcparadise added a lot of new functionality to this plugin here: https://github.com/chef/knife-opc/commit/c9f00fc48c8927e24f6342021236e4633f0534f7. It looks like your rebase of this PR is removing all the new functionality, which we'll want to keep. This is going to need some significant rework.

**Review by tas50 — CHANGES_REQUESTED** [review]
See comment above

**Comment by antima-gupta** [comment]
@tas50 I debugged the new functionality and found the code behavior is same. We have option `--remove-from-admin-groups` in user delete command so we don't need these changes.

**Comment by lamont-granquist** [comment]
merged knife-opc into chef/chef so closing this out, if its still relevant it needs to be moved:

https://github.com/chef/chef/pull/10187

**Pull request closed without merging**